target/cheri-common/cheri_tagmem.h

```diff
@@ -31,42 +31,299 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+
+#pragma once
+
 #include "qemu/osdep.h"
 #include "cpu.h"
 #include "exec/cpu-common.h"
+#include "exec/exec-all.h"
 #include "exec/memory.h"
+#include "cheri_tagmem_ex_locks.h"
 
 #if defined(TARGET_CHERI)
+
+/*
+ * TODO: I neatly pulled some qemu_tcg_mttcg_enabled branches into this header
+ * to ensure that when disabled extra locking nonsense compiles away, but
+ * apprently TARGET_SUPPORTS_MTTCG is the static thing, and
+ * qemu_tcg_mttcg_enabled is dynamic. Although qemu_tcg_mttcg_enabled
+ * will default to false if TARGET_SUPPORTS_MTTCG is
+ * undefined, it can be still be overridden on the command line. I think
+ * on CHERI I am happy to say this should be unsupported, and use
+ * TARGET_SUPPORTS_MTTCG here. For now just keeping as is. This means
+ * that even if we compile without TARGET_SUPPORTS_MTTCG there will be some
+ * extra branches to skip lock taking / releases.
+ */
```

Comment:
I don't see this as a particularly worthwhile comment? The current approach is the right one.